File: RootTools2/src/main/java/com/rarnu/tools/neo/activity/BuildPropEditActivity.java
package com.rarnu.tools.neo.activity;

import android.app.Fragment;
import android.content.res.Configuration;
import android.os.Bundle;
import com.rarnu.tools.neo.base.BaseDialog;
import com.rarnu.tools.neo.fragment.BuildPropEditFragment;

public class BuildPropEditActivity extends BaseDialog {

    private static int screenState1 = -1, screenState2 = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        if (screenState1 == -1 && screenState2 == -1) {
            screenState1 = screenState2 = getResources().getConfiguration().orientation;
        } else {
            screenState1 = screenState2;
            screenState2 = getResources().getConfiguration().orientation;
        }
        if (screenState1 == Configuration.ORIENTATION_LANDSCAPE && screenState2 == Configuration.ORIENTATION_PORTRAIT) {
            screenState1 = screenState2 = -1;
            finish();
            return;
        }
    }

    @Override
    public boolean getCondition() {
        return false;
    }

    @Override
    public Fragment replaceFragment() {
        return new BuildPropEditFragment();
    }

    @Override
    public int customTheme() {
        return 0;
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/activity/BuildPropEditActivity.kt
package com.rarnu.tools.neo.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.rarnu.android.toEditable
import com.rarnu.tools.neo.R
import com.rarnu.tools.neo.data.BuildPropInfo
import kotlinx.android.synthetic.main.fragment_fakedev_edit.*

class BuildPropEditActivity : Activity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_fakedev_edit)

        btnCancel.setOnClickListener(this)
        btnSave.setOnClickListener(this)

        val item = intent.getSerializableExtra("item") as BuildPropInfo?
        tvHead.text = item?.buildName
        etValue.text = item?.buildValue?.toEditable()

    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btnSave -> {
                setResult(RESULT_OK, Intent().apply {
                    putExtra("item", BuildPropInfo(tvHead.text.toString(), etValue.text.toString()))
                    putExtra("position", intent.getIntExtra("position", -1))
                })
                finish()
            }
            R.id.btnCancel -> finish()
        }
    }

}
